### ISSUE TITLE
Keras(Python) stops working after first epoch with any error

### ISSUE THREAD

**Issue by Oxi84** [keras-team/keras#9497]
Hello,
I have latest version of theano and keras and I am trying to run script from keras from this page for learning purposes:
https://machinelearningmastery.com/develop-word-based-neural-language-models-python-keras/

This first example of full code from the page.

When I run the code it work ok untill it starts training the model. Then python just stops executing without reporting any error. What could be the problem?

########################################################
Using Theano backend.
WARNING (theano.tensor.blas): Using NumPy C-API based implementation for BLAS functions.
Vocabulary Size: 22
Total Sequences: 24
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
embedding_1 (Embedding)      (None, 1, 10)             220       
_________________________________________________________________
lstm_1 (LSTM)                (None, 50)                12200     
_________________________________________________________________
dense_1 (Dense)              (None, 22)                1122      
=================================================================
Total params: 13,542
Trainable params: 13,542
Non-trainable params: 0
_________________________________________________________________
None
Epoch 1/500
###############################################

Thanks

**Comment by fchollet** [comment]
Sounds like a segfault (or similar) within Theano itself.

**Comment by Oxi84** [comment]
Thanks for the answer. Is there any way to debug and determine what the problem is? Because I do not get any error - such as segfault error. It just stops working.

Is there any way to turn on log or something.

Also, does anyone have any working short example of using LSTM example in Keras, so I can check if the example works for me.

**Comment by fchollet** [comment]
You will find plenty of LSTM examples in the Keras examples folder.
Since the issue is with Theano, I recommend you try using the TensorFlow backend instead.

**Comment by Oxi84** [comment]
I prefer Theano because I run python 2.7 and for Tensorflow I need 3.4+ (I think). So it would be the best if I managed for the Theano to work somehow.

Thanks I have found the example and will try some of them.
I tried another example not in the keras folder and I got this error:
WARNING (theano.gof.compilelock): Overriding existing lock by dead process '2092' (I am process '7072')

**Comment by Oxi84** [comment]
I tried a few examples and I have found that this example behaves the same way: - https://github.com/keras-team/keras/blob/master/examples/imdb_lstm.py

It stops working after the first epoch as well:

**Comment by fchollet** [comment]
It sounds like you have a bad installation of Theano. Uninstall everything,
reinstall.

Or alternatively, use TensorFlow (which works with Python 2.7, too).

**Comment by Oxi84** [comment]
Will try to uninstall it. Ordinary ANNs work fine with such Theano - just LSTM do not work for soem reason.

I use windows as well so I guess it is not possible to install it without complications: - https://stackoverflow.com/questions/45316569/how-to-install-tensorflow-on-python-2-7-on-windows

**Comment by Oxi84** [comment]
I did:
pip uninstall theano
pip uninstall keras

pip install keras
pip install Theano

Still I get the same error. On another PC of mine it works well. But this one is much slower.

**Comment by Oxi84** [comment]
Also I get Windows error in event viewer - seems like faulty module is Theano:

**Comment by Oxi84** [comment]
Woks now but slow - i got this warning: WARNING (theano.configdefaults): g++ not detected ! Theano will be unable to execute optimized C-implementations (for both CPU and GPU) and will default to Python implementations. Performance will be severely degraded. To remove this warning, set Theano flags cxx to an empty string.
WARNING (theano.tensor.blas): Using NumPy C-API based implementation for BLAS functions.

Is there any way to maanually set gc++ compiler. I installed MinGW, but theano still doesn't detect it. I would like to put the patch to the .exe somewhere.


Solution was:I reinstalled c+ compiler, it was cygwin (TMD or something ) and I have deinstalled it.  This was error cause.

Now works, but it is really slow.
